### DISCUSSION TITLE
What is ClusterCacheTracker ?

### DISCUSSION THREAD

**Discussion by pierreozoux** [kubernetes-sigs/cluster-api#10164] [Q&A] [answered]
Since 1.6.1, I see this line in my logs:
```
I0216 09:15:17.687758       1 cluster_cache_tracker.go:164] "Couldn't find controller pod metadata, the ClusterCacheTracker will always access clusters using the regular apiserver endpoint" component="remote/clustercachetracker"
```

What is ClusterCacheTracker? What is the benefit of using it?
How can I activate it?
Is there some risks to activate it? I see some issues related to it.

Thanks a lot for your help!

Have a nice day!

**Comment by sbueringer** [comment, marked as answer]
ClusterCacheTracker is a component that CAPI components use to create and cache a controller-runtime client to communicate with workload clusters.

The log line you see indicates that your controller Pod does not specify these env vars: https://github.com/kubernetes-sigs/cluster-api/blob/500fc1fec6867a5b27b7bda38882f7d1957a8b52/config/manager/manager.yaml#L30-L41.

The env vars are used for an optimization so we directly talk to the kubernetes.default.svc service if the workload cluster is the cluster the controller is running on. The ClusterCacheTracker will be used in any case, independent of if we can use this optimization or not.

(I'm not sure what your controller is, but CAPI components have been using the ClusterCacheTracker since a few years)

**Reply by pierreozoux** [reply]
Thanks for the detailed explanation!
